### PR TITLE
(MODULES-1700) Change the filebucketing behavior so static_compiler can ...

### PR DESCRIPTION
**Pull request by woneill**
...handle them

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -141,7 +141,7 @@
   }
 
   File {
-    backup  => false,
+    backup  => $backup,
```
Comment (hunner) [inline manifests/init.pp]: I don't think this is needed to work with the static compiler. None of these files have sources pointing at the puppet fileserver.

File: manifests/setup.pp [hunk 2]
```diff
@@ -46,10 +46,6 @@
     default   => $script_path
   }
 
-  File {
-    backup => false,
-  }
```
Comment (hunner) [inline manifests/setup.pp]: Instead of changing the default, could you just change the backup value for the `File[$script_path]` resource?

Comment (jhoblitt) [inline manifests/setup.pp]: Resource defaulting has caused so many problematic side effects I'm happy to see it go...

**Comment by hunner** [comment]
I merged some things and now it needs a rebase, also.

File: manifests/init.pp (hunk 1 above)
Comment (woneill) [inline manifests/init.pp]: I tried reverting just this line and it looks like concat::setup inherited it:
```
Error: /Stage[main]/Concat::Setup/File[/var/lib/puppet/concat/bin/concatfragments.rb]/ensure: change from absent to file failed: Could not retrieve content for {md5}b684db0eac243553a6a79365119a363d from filebucket
```

Comment (hunner) [inline manifests/init.pp]: Okay, it's fine then.

File: manifests/setup.pp (hunk 2 above)
Comment (woneill) [inline manifests/setup.pp]: I couldn't come up with a way to get the value of $backup inside concat::setup short of making it a parameterized class. Plus it seems that this is being inherited anyway

**Comment by woneill** [comment]
I'll handle the rebase and update

**Comment by woneill** [comment]
I rebased then did a `git push --force`

**Pull request merged**